lib/plugins/aws/provider/awsProvider.js

```diff
@@ -237,6 +237,85 @@ class AwsProvider {
             },
           },
         },
+        resources: {
+          properties: {
+            AWSTemplateFormatVersion: {
+              type: 'string',
+            },
+            Description: {
+              type: 'string',
+            },
+            Metadata: {
+              type: 'object',
+            },
+            Parameters: {
+              type: 'object',
+            },
+            Mappings: {
+              type: 'object',
+            },
+            Conditions: {
+              type: 'object',
+            },
+            Transform: {
+              type: 'object',
+            },
+            // Not replicating the full JSON schema from https://s3.amazonaws.com/cfn-resource-specifications-us-east-1-prod/schemas/2.15.0/all-spec.json
+            // as that gets into the specifics for each resource type.
+            //
+            // The only required attribute is `Type`; `Properties` and other common attributes are optional.
+            // See also https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html
+            Resources: {
+              type: 'object',
+              patternProperties: {
+                '^[a-zA-Z0-9]{1,255}$': {
+                  type: 'object',
+                  properties: {
+                    Type: { type: 'string' },
+                    Properties: { type: 'object' },
+                    CreationPolicy: { type: 'object' },
+                    DeletionPolicy: { not: { type: 'null' } },
+                    DependsOn: { type: 'array', items: { not: { type: 'null' } } },
+                    Metadata: { type: 'object' },
+                    UpdatePolicy: { type: 'object' },
+                    UpdateReplacePolicy: { not: { type: 'null' } },
```

Comment:
`DeletionPolicy` and `UpdateReplacePolicy` are defined as `string`, but I'm not sure if you're allowed to reference parameters for them, so I'm being really permissive here. `null` is definitely not a correct value, so will flag that.

I suppose another option would be `{ oneOf: [{type: 'object'},{type: 'string'}] }`. Thoughts?

Reply:
I assume only `string` type is supported in them. It's very _meta_ property (as e.g. `Type`, afaik, you cannot define it by referencing some other resource).

At least documentation doesn't list any other configuration options: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html

I'd stick to `string` (unless we have some proof it can work with other type of value)